### ISSUE TITLE
Code: no errors shown if server is not executable

### ISSUE THREAD

**Issue by lnicola** [rust-analyzer/rust-analyzer#4169]
1. set `rust-analyzer.serverPath` to `~/.cargo/bin/rust-analyzer`
2. `chmod -x ~/.cargo/bin/rust-analyzer`
3. open a Rust project in Code

The extension doesn't work and no errors are reported:

![image](https://user-images.githubusercontent.com/308347/80338808-f2837f80-8865-11ea-879c-47e3721a8bdf.png)

![image](https://user-images.githubusercontent.com/308347/80338819-f7483380-8865-11ea-9814-b0d24e905db1.png)

CC @Veetaha

**Comment by eminence** [comment]
I don't have a linux machine with vscode to test, but I would have thought that [this code](https://github.com/rust-analyzer/rust-analyzer/blob/7a9ba1657daa9fd90c639dcd937da11b4f526675/editors/code/src/main.ts#L182-L187) would have caught this problem:

```ts
    const res = spawnSync(path, ["--version"], { encoding: 'utf8' });
    log.debug("Checked binary availability via --version", res);
    log.debug(res, "--version output:", res.output);
    if (res.status !== 0) {
        throw new Error(`Failed to execute ${path} --version`);
    }
```

**Comment by Veetaha** [comment]
This is the field of responsibility of `vscode-languageclient` it is passed the path to the executable and it invokes it on its own. https://github.com/rust-analyzer/rust-analyzer/blob/7a9ba1657daa9fd90c639dcd937da11b4f526675/editors/code/src/client.ts#L12-L15

We could certainly check some invariant before that, but I am not sure it is worth it since this is the case of user-specified-path and the user is in charge of keeping the invariants here. However, I won't argue that there could be a good error message for this, maybe we just ignore it somewhere during the client creation, or `vscode-languageclient` does it itself...

**Comment by lnicola** [comment]
It might be a swallowed exception, but I think some message should be shown, otherwise it's too confusing for the user.

I ran into this after I downloaded a binary release and forgot to run chmod.

**Comment by lnicola** [comment]
We now get:

```
INFO [22/01/2021, 20:17:09]: Using server binary at rust-analyzer
WARN [22/01/2021, 20:17:09]: rust-analyzer --version: {
  error: [Error: spawnSync rust-analyzer EACCES
  	at Object.spawnSync (internal/child_process.js:1041:20)
  	at Object.spawnSync (child_process.js:616:24)
  	at Object.isValidExecutable (/home/grayshade/.vscode/extensions/matklad.rust-analyzer-0.4.0-dev/out/src/main.js:17275:40)
  	at bootstrapServer (/home/grayshade/.vscode/extensions/matklad.rust-analyzer-0.4.0-dev/out/src/main.js:21147:15)
  	at async bootstrap (/home/grayshade/.vscode/extensions/matklad.rust-analyzer-0.4.0-dev/out/src/main.js:21084:18)
  	at async tryActivate (/home/grayshade/.vscode/extensions/matklad.rust-analyzer-0.4.0-dev/out/src/main.js:21004:24)
  	at async activate (/home/grayshade/.vscode/extensions/matklad.rust-analyzer-0.4.0-dev/out/src/main.js:20978:5)
  	at async Promise.all (index 0)] {
    errno: 'EACCES',
    code: 'EACCES',
    syscall: 'spawnSync rust-analyzer',
    path: 'rust-analyzer',
    spawnargs: [ '--version' ]
  },
```
